### PR TITLE
Fix registry client

### PR DESCRIPTION
**Pull request by lvfxx**
Fixes to make clientInfo NetworkServiceRegistryClient tests work

### CODE REVIEW

File: pkg/registry/core/next/registry_client.go [hunk 1]
```diff
@@ -49,26 +49,30 @@ func NewWrappedRegistryClient(wrapper RegistryClientWrapper, clients ...registry
...
-	return n.clients[n.index].RegisterNSE(withNextRegistryClient(ctx, nil), request, opts...)
+	return n.clients[n.index].RegisterNSE(withNextRegistryClient(ctx, &tailRegistryClient{}), request, opts...)
 }
 
 func (n *nextRegistryClient) BulkRegisterNSE(ctx context.Context, opts ...grpc.CallOption) (registry.NetworkServiceRegistry_BulkRegisterNSEClient, error) {
 	if n.index+1 < len(n.clients) {
 		return n.clients[n.index].BulkRegisterNSE(withNextRegistryClient(ctx, &nextRegistryClient{clients: n.clients, index: n.index + 1}), opts...)
 	}
-	return n.clients[n.index].BulkRegisterNSE(withNextRegistryClient(ctx, nil), opts...)
+	return n.clients[n.index].BulkRegisterNSE(withNextRegistryClient(ctx, &tailRegistryClient{}), opts...)
```
Comment (edwarnicke) [inline pkg/registry/core/next/registry_client.go]: Move the &tailRegistryClient{} to context.go (see https://github.com/networkservicemesh/sdk/pull/106 for the example for chaining networkservice).

Comment (edwarnicke) [reply pkg/registry/core/next/registry_client.go]: This makes it safe to run a chain element even its not in a chain.

Comment (lvfxx) [reply pkg/registry/core/next/registry_client.go]: Ok, did it

File: pkg/registry/core/next/tail_registry_client.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/registry"
+	"google.golang.org/grpc"
+)
+
+// tailRegistryClient is a simple implementation of registry.NetworkServiceRegistryClient that is called at the end
+// of a chain to ensure that we never call a method on a nil object
+type tailRegistryClient struct{}
```
Comment (denis-tingajkin) [inline pkg/registry/core/next/tail_registry_client.go]: Do we need the same thing for `next.RegistryServer`?
I guess we need to check that `pkg/registry/next` for `RegistryClient`/`RegistryServer` is working as expected.
Let's cover `pkg/registry/next` by unit tests and fix problems.

Comment (lvfxx) [reply pkg/registry/core/next/tail_registry_client.go]: Yes, we need. I've added it

File: pkg/registry/core/next/registry_server.go [hunk 3]
```diff
@@ -47,8 +47,11 @@ func NewWrappedRegistryServer(wrapper RegistryServerWrapper, servers ...registry
 }
 
 // NewRegistryServer - creates a chain of servers
-func NewRegistryServer(server []registry.NetworkServiceRegistryServer) registry.NetworkServiceRegistryServer {
-	return NewWrappedRegistryServer(nil, server...)
+func NewRegistryServer(servers ...registry.NetworkServiceRegistryServer) registry.NetworkServiceRegistryServer {
+	if len(servers) == 0 {
```
Comment (edwarnicke) [inline pkg/registry/core/next/registry_server.go]: Nice catch!

**Comment by edwarnicke** [comment]
Can we move  pkg/registry/core/next/tests/ up to  pkg/registry/core/next/ using package 'next_test' in keeping with standard go idioms?

**Pull request merged**